libs/langchain/langchain/document_transformers/dedup.py

```diff
@@ -0,0 +1,36 @@
+from typing import Any, Iterable, Iterator, Sequence, Union
+
+from langchain.schema import BaseDocumentTransformer, Document
+from langchain.schema.document import _deduplicate_in_order, _HashedDocument
+
+
+class DedupDocumentTransformer(BaseDocumentTransformer):
```

Comment:
Perhaps add a unit test? chat gpt can probably write it if given the relevant interfaces